src/js/jobs/components/JobArgs.tsx

```diff
@@ -0,0 +1,191 @@
+
+/**
+ * A single row of the job arguments table.
+ *
+ * @param children - What to display as the value of the argument
+ * @param title - The name of the job argument
+ * @param className - An optional class name to apply to the row
+ * @returns A table row containing the argument name and value
+ */
+function JobArgsRow({ children, title, className }: { children: ReactNode; title: string; className?: string }) {
+    return (
+        <tr className={className}>
+            <th>{title}</th>
+            <td>{children}</td>
+        </tr>
+    );
+}
+
+/**
+ * Rows showing important arguments when running a sample analysis workflow.
+ *
+ * @param args - The complete list of arguments used to run the job
+ * @returns Table rows containing the arguments of the job
+ */
+export function AnalysisRows({ sample_id, analysis_id }: { sample_id: string; analysis_id: string }) {
+    return (
+        <>
+            <JobArgsRow title="Sample">
+                <Link to={`/samples/${sample_id}`}>{sample_id}</Link>
+            </JobArgsRow>
+            <JobArgsRow title="Analysis">
+                <Link to={`/samples/${sample_id}/analyses/${analysis_id}`}>{analysis_id}</Link>
+            </JobArgsRow>
+        </>
+    );
+}
+
+/**
+ * Rows showing important arguments when running an "build_index" workflow.
+ *
+ * @param args - The complete list of arguments used to run the job
+ * @returns Table rows containing the arguments of the job
+ */
+export function BuildIndexRows({ index_id, ref_id }: { index_id: string; ref_id: string }) {
+    return (
+        <>
+            <JobArgsRow title="Reference">
+                <Link to={`/refs/${ref_id}`}>{ref_id}</Link>
+            </JobArgsRow>
+            <JobArgsRow title="Index">
+                <Link to={`/refs/${ref_id}/indexes/${index_id}`}>{index_id}</Link>
+            </JobArgsRow>
+        </>
+    );
+}
+
+/**
+ * Rows showing important arguments when running an "create_sample" workflow.
+ *
+ * @param args - The complete list of arguments used to run the job
+ * @returns Table rows containing the arguments of the job
+ */
+export function CreateSampleRows({ sample_id }: { sample_id: string }) {
+    return (
+        <JobArgsRow title="Sample">
+            <Link to={`/samples/${sample_id}`}>{sample_id}</Link>
+        </JobArgsRow>
+    );
+}
+
+/**
+ * Rows showing important arguments when running an "create_subtraction" workflow.
+ *
+ * @param args - The complete list of arguments used to run the job
+ * @returns Table rows containing the arguments of the job
+ */
+export function CreateSubtractionRows({ subtraction_id }: { subtraction_id: string }) {
+    return (
+        <JobArgsRow title="Subtraction">
+            <Link to={`/subtractions/${subtraction_id}`}>{subtraction_id}</Link>
+        </JobArgsRow>
+    );
+}
+
+/**
+ * Rows showing important arguments when running an "update_sample" workflow.
+ *
+ * @param args - The complete list of arguments used to run the job
```

Comment:
This may have happened while you were away...

I found a lot of opinions online saying that documenting the props `type` for the component in TS is better than using `@param`.

I have to agree. This one is a good example where `@param` line is a bit distant from the function signature and is less descriptive than a type would be.